s3-lambda-bedrock-js-sam/README.md

```diff
@@ -0,0 +1,72 @@
+
+Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd s3-lambda-bedrock-js-sam
+    ```
+1. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+    ```
+    sam deploy --guided
+    ```
+1. During the prompts:
+    * Enter a stack name
+    * Enter the desired AWS Region, just make sure you have access to the Titan Text Embeddings V1 model in the region
+    * Enter value of the DocumentBucketName parameter which will be the name of created S3 bucket
+    * Allow SAM CLI to create IAM roles with the required permissions.
+
+    Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+
+2. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
+## How it works
+
+When a new object is uploaded to an S3 bucket, S3 detects the putObject event and invokes a Lambda function asynchronously, passing details about the uploaded object. The Lambda function then retrieves the object's content and sends it to the Amazon Bedrock Titan Embedding service, which generates vector embeddings representing the content's meaning and context.
+
+## Testing
+
+To test the system, we should first tail the logs of the Lambda function. Then, we can upload the sample.txt to our designated S3 bucket. Once the document is uploaded, we can observe the logs to verify the Lambda function's execution and the embeddings of the uploaded document. 
+
+1. Tail to the Lambda function log to get the logs
+```
+aws logs tail "/aws/lambda/DocumentEmbeddingFunction" --follow --format json 
```

Comment:
The function name ```DocumentEmbedding``` in CloudFormation output would not match ```DocumentEmbeddingFunction ```

Reply:
Made the changes. Corrected the name to DocumentEmbedding.